lib/streamlit/elements/widgets/button.py

```diff
@@ -744,6 +768,20 @@ def _page_link(
                 f"Could not find page: `{page}`. Must be the file path relative to the main script, from the directory: `{os.path.basename(main_script_directory)}`. Only the main app file and files in the `pages/` directory are supported."
             )
 
+        if query_params is not None:
+            if not hasattr(query_params, "get_all"):
```

Comment:
Why is there this check out of curiosity? I assume it's because to check that we can use it to do `from_dict` but maybe could use a comment here why this check.

Reply:
If the `query_params` has a `get_all` method we assume (perhaps incorrectly) that it is a QueryParams object or some derivative. It's kind of a kludgy check, since we never actually _call_ `get_all` at all -- we call `to_string()` instead, but checking for `to_string()` seems like it would open us up to unintended behavior.

Perhaps this should be an `isinstance` check, and ONLY special-case objects that are `QueryParams` or `QueryParamsProxy`s?